components/CommentList/index.js

```diff
@@ -1,52 +1,84 @@
 import SVGIcon from "@/components/SVGIcon";
-import Image from "next/image";
+import { useState } from "react";
 import styled from "styled-components";
+import { StyledDate } from "../PostingDetails/StyledPostingDetails";
 import ProfilLink from "../ProfilLink";
+import ProfileImage from "../ProfileDetails/ProfileImage";
+import Button from "../Button";
+import { StyledPostingPrevHeader } from "../PostingPreview/StyledPostingPreview";
 
 export default function CommentList({
   comments,
   onDeleteComment,
   loggedInUserID,
   postIsFromUser,
 }) {
+  const [commentCount, setCommentCount] = useState(5);
+
+  const reverseComments = [...comments].reverse().slice(0, commentCount);
+
   return (
-    <section>
-      {comments.length ? (
-        comments.map((comment) => {
+    <StyledCommentList>
+      {reverseComments.length > 0 &&
+        reverseComments.map((comment) => {
           return (
             <StyledComment key={comment._id}>
-              <ProfilLink href={`/users/${comment.user._id}`}>
-                <Image
-                  src={comment.user.image}
-                  width={30}
-                  height={30}
-                  alt={`profile picture of ${comment.user.name}`}
-                />
-                <div>
-                  <p>{comment.user.name}</p>
-                  <p>{comment.date}</p>
-                </div>
-              </ProfilLink>
+              <StyledPostingPrevHeader>
+                <ProfilLink href={`/users/${comment.user._id}`}>
+                  <ProfileImage
+                    userName={comment.user.name}
+                    image={comment.user.image}
+                    width={30}
+                  />
+                  <div>
+                    <p>{comment.user.name}</p>
+                    <StyledDate>{comment.date}</StyledDate>
+                  </div>
+                </ProfilLink>
+                {(comment.user._id === loggedInUserID || postIsFromUser) && (
+                  <Button
+                    variant="deleteComment"
+                    type="button"
+                    onClick={() => onDeleteComment(comment._id)}
+                  >
+                    <SVGIcon variant="bin" width="13" />
+                  </Button>
+                )}
+              </StyledPostingPrevHeader>
 
-              <p>{comment.content}</p>
-              {(comment.user._id === loggedInUserID || postIsFromUser) && (
-                <button
-                  type="button"
-                  onClick={() => onDeleteComment(comment._id)}
-                >
-                  <SVGIcon variant="bin" width="13" />
-                </button>
-              )}
+              <StyledCommentContent>{comment.content}</StyledCommentContent>
             </StyledComment>
           );
-        })
+        })}
+      {comments.length > 5 ? (
```

Comment:
```suggestion
      {comments.length > 5 && (
```

and then leave the empty fragments out.